pkgs/development/tools/misc/binutils/default.nix

```
@@ -106,7 +104,15 @@ stdenv.mkDerivation {
       else ./mips64-default-n64.patch)
   ;
 
-  outputs = [ "out" "info" "man" ];
+  outputs = [ "out" "info" "man" "dev" ]
+  # Ideally we would like to always install 'lib' into a separate
+  # target. Unfortunately cross-compiled binutils installs libraries
+  # across both `$lib/lib/` and `$out/$target/lib` with a reference
+  # from $out to $lib. Probably a binutils bug: all libraries should go
+  # to $lib as binutils does not build target libraries. Let's make our
+  # life slightly simpler by installing everything into $out for
+  # cross-binutils.
+  ++ lib.optionals (targetPlatform == hostPlatform) [ "lib" ];
```

```suggestion
  ;
```

Can we keep the layout the same between native and cross?  The big promise of CA-derivations is being able to swap them (for example if you have a slow embedded platform you can cross-build most of its packages, and native-build only the ones that don't support cross-compilation).

---

Yeah, I'd like to fix it upstream first as `binutils` does not seem to install any target-specific libraries outside `lib/`. The split between `libbfd.so` and `libsframe.so` locations does not look intentional. Once it's sorted we could bring both cases closer (either by using or not by using target prefixes consistently).

I can understand why FHS system wants to distinguish two `binutils` installs for two different targets in a single prefix. Maybe it's worth fixing as well.